libraries/AP_Notify/AP_Notify.cpp

```diff
@@ -244,27 +224,23 @@ void AP_Notify::add_backends(void)
     ADD_BACKEND(new DiscoLED());
     ADD_BACKEND(new ToneAlarm_Linux());
 
-  #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RST_ZYNQ
-    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_EXTERNAL));
-
   #else // other linux
     ADD_BACKEND(new AP_BoardLED());
-    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_EXTERNAL));
-    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_INTERNAL));
     ADD_BACKEND(new ToneAlarm_Linux());
   #endif
 
 #elif CONFIG_HAL_BOARD == HAL_BOARD_F4LIGHT
-    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_EXTERNAL));
     ADD_BACKEND(new Display());
     ADD_BACKEND(new Buzzer());
     ADD_BACKEND(new AP_BoardLED2()); // needs AP_BoardLED2 in master
 #else
     ADD_BACKEND(new AP_BoardLED());
-    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_EXTERNAL));
-    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_INTERNAL));
     ADD_BACKEND(new Display());
 #endif
+
+    // All backends probe for these devices
+    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_EXTERNAL));
+    ADD_BACKEND(new ToshibaLED_I2C(TOSHIBA_LED_I2C_BUS_INTERNAL));
```

Comment:
It won't have a side-effect, but I'm not sure we should go in this direction. I think we should always probe for devices on the *external* bus, i.e. a bus the user can plug things on. For the internal, I'd rather leave for each board to define what it has.

And not all drivers are actually prepared for checking if the device is there. This one in particular does not. If any device has a 0x55 address it will be treated as toshiba led. With the limited set of devices we support in ardupilot, it can already conflict - it could be a PCA9685.

Reply:
The only reason I added the internal probing is that for some devices (Pixhawk) it's valid, and then on some carrier boards both the internal and external buses have been exposed.

The other thought I had was to add a parameter that held a bitmask of notify LED's to probe for with the default set appropriately in the firmware. The attraction to this is I actually want to use the PCA9685 on a FMUv3 board, the downside to this is that it means all the drivers would be linked (except for ones where we physically don't have the pins defined like AP_BoardLED witthout a 3rd GPIO LED).